pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go

```diff
@@ -209,11 +209,16 @@ func TestController_syncConfigurations(t *testing.T) {
 		assert.Equal(t, 0, fakeController.queue.Len())
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
+		signCh := make(chan struct{})
+		defer func() {
+			<-signCh
+		}()
```

Comment:
if the next line returns an error, the code will block forever?

Reply:
Moved the defer func below the watcher error check. Thanks!